src/components/DataDimension/DataElementSelector.js

```diff
@@ -0,0 +1,321 @@
+                    : null}
+            </SingleSelectField>
+            <style jsx>{styles}</style>
+        </div>
+    )
+}
+
+GroupSelector.propTypes = {
+    currentValue: PropTypes.string.isRequired,
+    displayNameProp: PropTypes.string.isRequired,
+    onChange: PropTypes.func.isRequired,
+}
+
+const DisaggregationSelector = ({ currentValue, onChange }) => {
+    const options = getOptions()
+    return (
+        <div className="group-select">
+            <SingleSelectField
+                selected={currentValue}
+                onChange={(ref) => onChange(ref.selected)}
+                dense
+                dataTest={'data-element-disaggregation-select'}
+            >
+                {Object.entries(options).map((option) => (
+                    <SingleSelectOption
+                        value={option[0]}
+                        key={option[0]}
+                        label={option[1]}
+                    />
+                ))}
+            </SingleSelectField>
+            <style jsx>{styles}</style>
+        </div>
+    )
+}
+
+DisaggregationSelector.propTypes = {
+    currentValue: PropTypes.string.isRequired,
+    onChange: PropTypes.func.isRequired,
+}
+
+const DataElementSelector = ({ displayNameProp, onDoubleClick }) => {
+    const dataEngine = useDataEngine()
+
+    const [searchTerm, setSearchTerm] = useState('')
+    const [group, setGroup] = useState(DIMENSION_TYPE_ALL)
+    const [subGroup, setSubGroup] = useState(TOTALS)
+    const [options, setOptions] = useState([])
+    const [loading, setLoading] = useState(true)
+
+    const { isSorting } = useSortable({})
+
+    const rootRef = useRef()
+    const hasNextPageRef = useRef(false)
+    const searchTermRef = useRef(searchTerm)
+    const pageRef = useRef(0)
+    const filterRef = useRef({
+        dataType: DIMENSION_TYPE_DATA_ELEMENT,
+        group,
+        subGroup,
+    })
+
+    const fetchData = async (scrollToTop) => {
+        try {
+            setLoading(true)
+
+            const result = await apiFetchOptions({
+                dataEngine,
+                nameProp: displayNameProp,
+                filter: filterRef.current,
+                searchTerm: searchTermRef.current,
+                page: pageRef.current,
+            })
+
+            if (result?.dimensionItems) {
+                const newOptions = result.dimensionItems.map((item) => ({
+                    label: item.name,
+                    value: item.id,
+                    type: item.dimensionItemType,
+                    expression: item.expression,
+                }))
+
+                setOptions((prevOptions) =>
+                    pageRef.current > 1
+                        ? [...prevOptions, ...newOptions]
+                        : newOptions
+                )
+
+                setLoading(false)
+            }
+
+            hasNextPageRef.current = result?.nextPage ? true : false
+        } catch (error) {
+            // TODO handle errors
```

Comment:
@martinkrulltott  should a subtask be added for handling errors?

Reply:
Subtask: https://dhis2.atlassian.net/browse/DHIS2-15034